ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ip_access_list.py

```diff
@@ -0,0 +1,183 @@
+            ],
+        }
+        return outbound_access_list
+
+    def _get_access_list_inbound(self) -> dict | None:
+        interface_subnets = []
+        for l3_interface in self.shared_utils.l3_interfaces:
+            interface_ip = l3_interface.get("ip_address")
+            if interface_ip and interface_ip != "dhcp":
+                subnet = str(ip_network(interface_ip, False))
+                if subnet not in interface_subnets:
+                    interface_subnets.append(subnet)
+            # Need to see how to handle dhcp cases.
+
+        if not interface_subnets:
+            return {}
+
+        outbound_access_list = {
+            "name": "PUBLIC_TRANSPORT_IN",
+            "entries": [],
+        }
+        # Allow BGP
+        outbound_access_list["entries"].append(
+            {
+                "sequence": self._get_sequence(True),
+                "remark": "### Allow BGP ###",
+            },
+        )
+        for source in interface_subnets:
+            outbound_access_list["entries"].append(
+                {
+                    "sequence": self._get_sequence(),
+                    "action": "permit",
+                    "protocol": "tcp",
+                    "source": source,
+                    "destination": "any",
+                    "destination_ports_match": "eq",
+                    "destination_ports": ["bgp"],
+                },
+            )
+
+        # Allow GRE
+        outbound_access_list["entries"].append(
+            {
+                "sequence": self._get_sequence(),
+                "remark": "### Allow GRE ###",
+            },
+        )
+        for source in interface_subnets:
+            outbound_access_list["entries"].append(
+                {
+                    "sequence": self._get_sequence(),
+                    "action": "permit",
+                    "protocol": "gre",
+                    "source": source,
+                    "destination": "any",
+                },
+            )
+
+        # Allow IPSEC
+        outbound_access_list["entries"].append(
+            {
+                "sequence": self._get_sequence(),
+                "remark": "### Allow IPSEC ###",
+            },
+        )
+        for source in interface_subnets:
+            outbound_access_list["entries"].append(
+                {
+                    "sequence": self._get_sequence(),
+                    "action": "permit",
+                    "protocol": "udp",
+                    "source": source,
+                    "destination": "any",
+                    "destination_ports_match": "eq",
+                    "destination_ports": ["isakmp", "3478", "non500-isakmp"],
+                },
+            )
+
+        # Allow ICMP
+        outbound_access_list["entries"].append(
+            {
+                "sequence": self._get_sequence(),
+                "remark": "### Allow ICMP ###",
+            },
+        )
+        for source in interface_subnets:
```

Comment:
using this for loop multiple times, we can optimize this with some helper function or define a list for type of protocols then write this for loop under the protocol iteration loop.

Reply:
Sorry but my review was in more of a draft form.
Have updated it now. Please have a look if it looks good now.
Thanks